pkg/types/types.go

```diff
@@ -113,6 +113,14 @@ type DeployerWithPostTester interface {
 	PostTest(testErr error) error
 }
 
+// DeployerWithVersion allows the deployer to specify it's version
+type DeployerWithVersion interface {
+	Deployer
+
+	// Version determines the version of the deployer binary
+	Version() string
+}
+
```

Comment:
What would be involved in making `Version()` part of `Deployer`?  Seeing three different `DeployerWithFoo` interfaces makes me wonder if it keeps things simpler to fold into core

Reply:
I think that's something we can add in v0.2.0
Currently, all out of tree deployers won't have `Version()` and will break the core interface.